Tests/IntegrationTests.Shared/RelationshipTests.cs

```diff
@@ -295,56 +302,56 @@ public void TestExceptionsFromEmptyListOutOfRange()
             Dog scratch;  // for assignment in following getters
             Assert.Throws<IndexOutOfRangeException>(() => dani.Dogs.Insert(-1, bilbo));
             Assert.Throws<IndexOutOfRangeException>(() => dani.Dogs.Insert(0, bilbo));
-            Assert.Throws<IndexOutOfRangeException>(() => scratch = dani.Dogs[0]);
+            Assert.Throws<IndexOutOfRangeException>(() => scratch = dani.Dogs [0]);
         }
 
 
         [Test]
         public void TestExceptionsFromIteratingEmptyList()
         {
             var dani = realm.All<Owner>().Where(p => p.Name == "Dani").Single();
-            var iter =  dani.Dogs.GetEnumerator();
+            var iter = dani.Dogs.GetEnumerator();
             Assert.IsNotNull(iter);
             var movedOnToFirstItem = iter.MoveNext();
             Assert.That(movedOnToFirstItem, Is.False);
             Dog currentDog;
-            Assert.Throws<IndexOutOfRangeException>(() => currentDog = iter.Current );
+            Assert.Throws<IndexOutOfRangeException>(() => currentDog = iter.Current);
         }
 
 
         [Test]
         public void TestExceptionsFromTimsDogsOutOfRange()
         {
-            var tim = realm.All<Owner>().Single( p => p.Name == "Tim");
-            Assert.Throws<IndexOutOfRangeException>( () => tim.Dogs.RemoveAt(4) );
-            var bilbo = realm.All<Dog>().Single( p => p.Name == "Bilbo Fleabaggins");
+            var tim = realm.All<Owner>().Single(p => p.Name == "Tim");
+            Assert.Throws<IndexOutOfRangeException>(() => tim.Dogs.RemoveAt(4));
+            var bilbo = realm.All<Dog>().Single(p => p.Name == "Bilbo Fleabaggins");
             Dog scratch;  // for assignment in following getters
-            Assert.Throws<IndexOutOfRangeException>( () => tim.Dogs.Insert(-1, bilbo) );
-            Assert.Throws<IndexOutOfRangeException>( () => tim.Dogs.Insert(3, bilbo) );
-            Assert.Throws<IndexOutOfRangeException>( () => scratch = tim.Dogs[99] );
+            Assert.Throws<IndexOutOfRangeException>(() => tim.Dogs.Insert(-1, bilbo));
+            Assert.Throws<IndexOutOfRangeException>(() => tim.Dogs.Insert(3, bilbo));
+            Assert.Throws<IndexOutOfRangeException>(() => scratch = tim.Dogs [99]);
         }
 
         [Test]
         public void TestSettingStandAloneObjectToRelationship()
         {
-            var owner = realm.All<Owner>().ToList().First();
+            var owner = realm.All<Owner>().First();
             var dog = new Dog { Name = "Astro" };
-            
+
             using (var trans = realm.BeginWrite())
             {
                 owner.TopDog = dog;
                 trans.Commit();
             }
 
-            var dogAgain = realm.All<Dog>().Where(d => d.Name == "Astro").ToList().SingleOrDefault();
+            var dogAgain = realm.All<Dog>().Single(d => d.Name == "Astro");
             Assert.That(dogAgain, Is.Not.Null);
```

Comment:
Technically these assertions are now redundant as `Single` cannot return `null`. But it's not a big deal..